src/dual.jl

```diff
@@ -257,6 +257,14 @@ Base.copy(d::Dual) = d
 Base.eps(d::Dual) = eps(value(d))
 Base.eps(::Type{D}) where {D<:Dual} = eps(valtype(D))
 
+function Base.nextfloat(d::ForwardDiff.Dual{T,V,N}) where {T,V,N}
+    ForwardDiff.Dual{T}(nextfloat(d.value), d.partials)
+end
+
+function Base.prevfloat(d::ForwardDiff.Dual{T,V,N}) where {T,V,N}
+    ForwardDiff.Dual{T}(prevfloat(d.value), d.partials)
```

Comment:
```suggestion
    ForwardDiff.Dual{T}(prevfloat(float(d.value)), d.partials)
```

Reply:
On the other hand, e.g., `prevfloat(::Int)` is not defined, so maybe it's fine without `float(...)` and it should just error if `prevfloat(d.value)` is not defined?

Reply:
A code pattern can be
```julia
foo(x) = goo(float(x))
goo(x) = ...
```
Oh yeah. Yes, that's fine.